### PR TITLE
LOK-2666   Implement test cases to verify metrics threshold alert in Alert service  

### PR DESCRIPTION
**Pull request by Mohammad-Junid**
## Description
Implement test case when the threshold alert is triggered then alert should be created and cleared according to the flow in Alert service

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2666

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: alert/src/test/java/org/opennms/horizon/alertservice/stepdefs/BackgroundSteps.java [hunk 1]
```diff
@@ -87,4 +88,10 @@ public void kafkaRestServerURLInSystemProperty(String systemProperty) {
         log.info("Using Kafka base URL: {}", this.kafkaBootstrapUrl);
         kafkaTestHelper.setKafkaBootstrapUrl(kafkaBootstrapUrl);
     }
+
+    @Given("Kafka metrics threshold  topic {string}")
+    public void kafkaMetricsThresholdTopic(String tpoic) {
```
Comment (smunir-onms) [inline alert/src/test/java/org/opennms/horizon/alertservice/stepdefs/BackgroundSteps.java:93]: typo: tpoic -> topic

File: alert/src/test/java/org/opennms/horizon/alertservice/stepdefs/MonitorPolicySteps.java [hunk 2]
```diff
@@ -51,21 +51,11 @@
...
-import org.opennms.horizon.alerts.proto.EventType;
-import org.opennms.horizon.alerts.proto.ListAlertEventDefinitionsRequest;
-import org.opennms.horizon.alerts.proto.ManagedObjectType;
-import org.opennms.horizon.alerts.proto.MonitorPolicyList;
-import org.opennms.horizon.alerts.proto.MonitorPolicyProto;
-import org.opennms.horizon.alerts.proto.MonitorPolicySearchBy;
-import org.opennms.horizon.alerts.proto.OverTimeUnit;
-import org.opennms.horizon.alerts.proto.PolicyRuleProto;
-import org.opennms.horizon.alerts.proto.Severity;
+import org.opennms.horizon.alerts.proto.*;
```
Comment (smunir-onms) [inline alert/src/test/java/org/opennms/horizon/alertservice/stepdefs/MonitorPolicySteps.java]: please avoid wildcards imports

File: alert/src/test/java/org/opennms/horizon/alertservice/stepdefs/MonitorPolicySteps.java [hunk 3]
```diff
@@ -195,6 +194,7 @@ public void thePolicyIsCreatedInANewTenant() {
         try {
             log.info("Creating policy {}", policy);
             lastCreatedPolicy = grpcClient.getPolicyStub().createPolicy(policy);
+            System.out.println(policy.getRulesList().get(0).getEventType());
```
Comment (smunir-onms) [inline alert/src/test/java/org/opennms/horizon/alertservice/stepdefs/MonitorPolicySteps.java:210]: remove this System.out.println

File: alert/src/test/java/org/opennms/horizon/alertservice/stepdefs/MonitorPolicySteps.java [hunk 4]
```diff
@@ -481,4 +506,36 @@ private MonitorPolicySearchBy getMonitorPolicySearchBy(int pageSize, String colu
...
+        Optional<ThresholdMetricProto> thresholdMetricProtoOption = thresholdMetricData.asMaps().stream()
+                .map(data -> ThresholdMetricProto.newBuilder()
+                        .setName(data.get("name"))
+                        .setEnabled(Integer.parseInt(data.get("enabled")) == 1)
+                        .setThreshold(Float.parseFloat(data.get("threshold")))
+                        .setCondition(data.get("condition"))
+                        .setExpression(data.get("expression"))
+                        .build())
+                .findFirst();
+        ;
```
Comment (smunir-onms) [inline alert/src/test/java/org/opennms/horizon/alertservice/stepdefs/MonitorPolicySteps.java:522]: remove extra ;

File: alert/src/test/java/org/opennms/horizon/alertservice/stepdefs/MonitorPolicySteps.java (hunk 2 above)
Comment (Mohammad-Junid) [reply alert/src/test/java/org/opennms/horizon/alertservice/stepdefs/MonitorPolicySteps.java]: Done

File: alert/src/test/java/org/opennms/horizon/alertservice/stepdefs/AlertTestSteps.java [hunk 5]
```diff
@@ -473,4 +483,49 @@ public void verifyAlertWithUeiClearKeyIsMatchedReductionKeyOfAlertOfUie(String u
...
+                                        .setAlertName(data.get("alertname") + "-"
+                                                + alertConditionProto.get().getId())
+                                        .setStatus(alertStatus)
+                                        .build())
+                                .build())
+                        .toList())
+                .build();
+    }
+
+    @Then("send event  through kafka")
```
Comment (smunir-onms) [inline alert/src/test/java/org/opennms/horizon/alertservice/stepdefs/AlertTestSteps.java:527]: please remove the extra space [] "send event[] through kafka" and update references

**Review by smunir-onms — COMMENTED** [review]
Looks good to me, @aurang-zeb313 @cgorantla please review and approve this

**Review by cgorantla — APPROVED** [review]
LGTM!

**Pull request merged**